config/src/main/java/com/typesafe/config/impl/ConfigNodeObject.java

```diff
@@ -12,32 +12,83 @@
         super(children);
     }
 
-    protected ConfigNodeObject changeValueOnPath(Path desiredPath, AbstractConfigNodeValue value) {
+    public boolean hasValue(Path desiredPath) {
+        for (AbstractConfigNode node : children) {
+            if (node instanceof ConfigNodeField) {
+                Path key = ((ConfigNodeField) node).path().value();
+                if (key.equals(desiredPath) || key.startsWith(desiredPath)) {
+                    return true;
+                } else if (desiredPath.startsWith(key)) {
+                    if (((ConfigNodeField) node).value() instanceof ConfigNodeObject) {
```

Comment:
similarly here you could write `ConfigNodeObject obj = (ConfigNodeObject) field.value()` to avoid mixing cast noise into the logic below.